### DISCUSSION TITLE
click element only if visible and don't fail if not visible

### DISCUSSION THREAD

**Discussion by HemanthKu-Kore** [cypress-io/cypress#23237] [General]
How can we verify the element is visible before clicking on it AND if the element is not visible, the test should not fail and move to next step.

**Comment by stavares843** [comment]
hello! 👋 

have you tried this

```
// your cypress test code
it('does something different based on the class of the button', () => {
  // RERUN THIS TEST OVER AND OVER AGAIN
  // AND IT WILL SOMETIMES BE TRUE, AND
  // SOMETIMES BE FALSE.

  cy.get('button').then(($btn) => {
    if ($btn.hasClass('active')) {
      // do something if it's active
    } else {
      // do something else
    }
  })
})
``` 

as mentioned [here](https://docs.cypress.io/guides/core-concepts/conditional-testing#The-DOM-is-unstable)?

**Comment by HemanthKu-Kore** [comment]
@stavares843  cy.get('button') will fail if the DOM doesn't have that element.

I need something like this in cypress. Below is a selenium code in java.
```
WebDriverWait w = new WebDriverWait(driver,3);
try {
   w.until(ExpectedConditions.presenceOfElementLocated (By.cssSelector("button")));
}
catch(Exception e) {}
```

Cypress should wait until the element exists and visible AND then click on it. It should not fail if the element doesn't found.
